include/SimCore/KaonPhysics.h

```diff
@@ -0,0 +1,104 @@
+#ifndef SIMCORE_KAON_PHYSICS_H
+#define SIMCORE_KAON_PHYSICS_H
+
+#include <G4DecayTable.hh>
+#include <G4KaonMinus.hh>
+#include <G4KaonPlus.hh>
+#include <G4ParticleDefinition.hh>
+#include <G4VDecayChannel.hh>
+#include <G4VPhysicsConstructor.hh>
+#include <numeric>
+#include <vector>
+
+#include "Framework/Configure/Parameters.h"
+#include "Framework/Exception/Exception.h"
+
+namespace simcore {
+
+/**
+ * @class KaonPhysics
+ *
+ * @brief Allows the configuration of properties of kaons
+ * produced in the simulation, in particular setting their lifetime and
+ * branching ratios
+ *
+ * @note Only affects charged kaons, but similar changes could be added for the
```

Comment:
Right here, or in a separate class? Would it make sense to add neutrals while we're at it? If not, should we call this ChargedKaonPhysics? I'm trying to make sure future me is not confused by this and I can imagine several ways. My preference would be to deal with all kaons here if it is easy enough, and just toggle which kaons one would like to modify with a configuration parameter.

Reply:
Right, that's a good point. It won't be as "clean" with the neutral kaons since they don't have symmetric decay channels like the charged ones do but it is straight forward to do

Reply:
Should be there now!